<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Extract the backbone for the gene-geneset graph — ggs_backbone • GeneTonic</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Extract the backbone for the gene-geneset graph — ggs_backbone"><meta property="og:description" content="Extract the backbone for the gene-geneset graph, either for the genes or for the
genesets"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- dependencies from examples --><script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="libs/vis-9.1.0/vis-network.min.css" rel="stylesheet"><script src="libs/vis-9.1.0/vis-network.min.js"></script><script src="libs/visNetwork-binding-2.1.0/visNetwork.js"></script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GeneTonic</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.7.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/GeneTonic_manual.html">The `GeneTonic` User's Guide
</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/federicomarini/GeneTonic" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Extract the backbone for the gene-geneset graph</h1>
    <small class="dont-index">Source: <a href="https://github.com/federicomarini/GeneTonic/blob/HEAD/R/ggs_graph.R" class="external-link"><code>R/ggs_graph.R</code></a></small>
    <div class="hidden name"><code>ggs_backbone.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Extract the backbone for the gene-geneset graph, either for the genes or for the
genesets</p>
    </div>

    <div id="ref-usage">Usage,<div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">ggs_backbone</span><span class="op">(</span>
  <span class="va">res_enrich</span>,
  <span class="va">res_de</span>,
  annotation_obj <span class="op">=</span> <span class="cn">NULL</span>,
  gtl <span class="op">=</span> <span class="cn">NULL</span>,
  n_gs <span class="op">=</span> <span class="fl">15</span>,
  gs_ids <span class="op">=</span> <span class="cn">NULL</span>,
  bb_on <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"genesets"</span>, <span class="st">"features"</span><span class="op">)</span>,
  bb_method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sdsm"</span>, <span class="st">"fdsm"</span>, <span class="st">"hyperg"</span><span class="op">)</span>,
  bb_extract_alpha <span class="op">=</span> <span class="fl">0.05</span>,
  bb_extract_fwer <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"none"</span>, <span class="st">"bonferroni"</span>, <span class="st">"holm"</span><span class="op">)</span>,
  bb_fullinfo <span class="op">=</span> <span class="cn">FALSE</span>,
  bb_remove_singletons <span class="op">=</span> <span class="cn">TRUE</span>,
  color_graph <span class="op">=</span> <span class="cn">TRUE</span>,
  color_by_geneset <span class="op">=</span> <span class="st">"z_score"</span>,
  color_by_feature <span class="op">=</span> <span class="st">"log2FoldChange"</span>,
  <span class="va">...</span>
<span class="op">)</span></code></pre></div></div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>res_enrich</dt>
<dd><p>A <code>data.frame</code> object, storing the result of the functional
enrichment analysis. See more in the main function, <code><a href="GeneTonic.html">GeneTonic()</a></code>, to check the
formatting requirements (a minimal set of columns should be present).</p></dd>
<dt>res_de</dt>
<dd><p>A <code>DESeqResults</code> object.</p></dd>
<dt>annotation_obj</dt>
<dd><p>A <code>data.frame</code> object with the feature annotation
information, with at least two columns, <code>gene_id</code> and <code>gene_name</code>.</p></dd>
<dt>gtl</dt>
<dd><p>A <code>GeneTonic</code>-list object, containing in its slots the arguments
specified above: <code>dds</code>, <code>res_de</code>, <code>res_enrich</code>, and <code>annotation_obj</code> - the names
of the list <em>must</em> be specified following the content they are expecting</p></dd>
<dt>n_gs</dt>
<dd><p>Integer value, corresponding to the maximal number of gene sets to
be included</p></dd>
<dt>gs_ids</dt>
<dd><p>Character vector, containing a subset of <code>gs_id</code> as they are
available in <code>res_enrich</code>. Lists the gene sets to be included in addition to
the top ones (via <code>n_gs</code>)</p></dd>
<dt>bb_on</dt>
<dd><p>A character string, either "genesets" or "features", to specify which
entity should be based the backbone graph on.</p></dd>
<dt>bb_method</dt>
<dd><p>A character string, referring to the function to be called (
from the <code>backbone</code> package) for computing the backbone of the specified
bipartite graph. Defaults to "sdsm", as recommended in the <code>backbone</code> package.</p></dd>
<dt>bb_extract_alpha</dt>
<dd><p>A numeric value, specifying the significance level to
use when detecting the backbone of the network</p></dd>
<dt>bb_extract_fwer</dt>
<dd><p>A character string, defaulting to "none", specifying
which method to use for the multiple testing correction for controlling the
family-wise error rate</p></dd>
<dt>bb_fullinfo</dt>
<dd><p>Logical value, determining what will be returned as output:
either a simple <code>ìgraph</code> object with the graph backbone (if set to <code>FALSE</code>),
or a list object containing also the <code>backbone</code> object, and the gene-geneset
graph used for the computation (if <code>TRUE</code>)</p></dd>
<dt>bb_remove_singletons</dt>
<dd><p>Logical value, defines whether to remove or leave
in the returned graph the nodes that are not connected to other vertices</p></dd>
<dt>color_graph</dt>
<dd><p>Logical value, specifies whether to use information about
genesets or features to colorize the nodes, e.g. for this info to be used in
interactive versions of the graph</p></dd>
<dt>color_by_geneset</dt>
<dd><p>Character string, corresponding to the column in
<code>res_enrich</code> to be used for coloring the nodes if <code>bb_on</code> is set to "genesets".
Defaults to the "z_score", which can be obtained via <code><a href="get_aggrscores.html">get_aggrscores()</a></code></p></dd>
<dt>color_by_feature</dt>
<dd><p>Character string, corresponding to the column in
<code>res_de</code> to be used for coloring the nodes if <code>bb_on</code> is set to "features".
Defaults to the "log2FoldChange", which should be normally included in a
DESeqResults object.</p></dd>
<dt>...</dt>
<dd><p>Additional parameters to be passed internally</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>According to the <code>bb_fullinfo</code>, either a simple <code>ìgraph</code> object with
the graph backbone, or a named list object containing:</p><ul><li><p>the <code>igraph</code> of the extracted backbone</p></li>
<li><p>the <code>backbone</code> object itself</p></li>
<li><p>the gene-geneset graph used for the computation</p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"macrophage"</span><span class="op">)</span></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/mikelove/DESeq2" class="external-link">"DESeq2"</a></span><span class="op">)</span></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"org.Hs.eg.db"</span><span class="op">)</span></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/AnnotationDbi" class="external-link">"AnnotationDbi"</a></span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># dds object</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"gse"</span>, package <span class="op">=</span> <span class="st">"macrophage"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">dds_macrophage</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSet</a></span><span class="op">(</span><span class="va">gse</span>, design <span class="op">=</span> <span class="op">~</span> <span class="va">line</span> <span class="op">+</span> <span class="va">condition</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> using counts and average transcript lengths from tximeta</span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dds_macrophage</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html" class="external-link">substr</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dds_macrophage</span><span class="op">)</span>, <span class="fl">1</span>, <span class="fl">15</span><span class="op">)</span></span>
<span class="r-in"><span class="va">dds_macrophage</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">estimateSizeFactors</a></span><span class="op">(</span><span class="va">dds_macrophage</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> using 'avgTxLength' from assays(dds), correcting for library size</span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># annotation object</span></span>
<span class="r-in"><span class="va">anno_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span class="r-in">  gene_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dds_macrophage</span><span class="op">)</span>,</span>
<span class="r-in">  gene_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">mapIds</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>,</span>
<span class="r-in">    keys <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dds_macrophage</span><span class="op">)</span>,</span>
<span class="r-in">    column <span class="op">=</span> <span class="st">"SYMBOL"</span>,</span>
<span class="r-in">    keytype <span class="op">=</span> <span class="st">"ENSEMBL"</span></span>
<span class="r-in">  <span class="op">)</span>,</span>
<span class="r-in">  stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span class="r-in">  row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dds_macrophage</span><span class="op">)</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 'select()' returned 1:many mapping between keys and columns</span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># res object</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">res_de_macrophage</span>, package <span class="op">=</span> <span class="st">"GeneTonic"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">res_de</span> <span class="op">&lt;-</span> <span class="va">res_macrophage_IFNg_vs_naive</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># res_enrich object</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">res_enrich_macrophage</span>, package <span class="op">=</span> <span class="st">"GeneTonic"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">res_enrich</span> <span class="op">&lt;-</span> <span class="fu"><a href="shake_topGOtableResult.html">shake_topGOtableResult</a></span><span class="op">(</span><span class="va">topgoDE_macrophage_IFNg_vs_naive</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 500 gene sets in `topGOtableResult` object.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting for usage in GeneTonic...</span>
<span class="r-in"><span class="va">res_enrich</span> <span class="op">&lt;-</span> <span class="fu"><a href="get_aggrscores.html">get_aggrscores</a></span><span class="op">(</span><span class="va">res_enrich</span>, <span class="va">res_de</span>, <span class="va">anno_df</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">ggs_bbg</span> <span class="op">&lt;-</span> <span class="fu">ggs_backbone</span><span class="op">(</span><span class="va">res_enrich</span>,</span>
<span class="r-in">  <span class="va">res_de</span>,</span>
<span class="r-in">  <span class="va">anno_df</span>,</span>
<span class="r-in">  n_gs <span class="op">=</span> <span class="fl">50</span>,</span>
<span class="r-in">  bb_on <span class="op">=</span> <span class="st">"genesets"</span>,</span>
<span class="r-in">  color_graph <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span class="r-in">  color_by_geneset <span class="op">=</span> <span class="st">"z_score"</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> This matrix object looks like an unweighted bipartite network of 50 agents and 319 artifacts.</span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">ggs_bbg</span><span class="op">)</span></span>
<span class="r-plt img"><img src="ggs_backbone-1.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># if desired, one can also plot the interactive version</span></span>
<span class="r-in"><span class="fu">visNetwork</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/visNetwork/man/visNetwork-igraph.html" class="external-link">visIgraph</a></span><span class="op">(</span><span class="va">ggs_bbg</span><span class="op">)</span></span>
<div id="htmlwidget-c19f8428d06c7fe05c86" style="width:700px;height:432.632880098888px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-c19f8428d06c7fe05c86">{"x":{"nodes":{"id":["DNA replication checkpoint","DNA replication initiation","G protein-coupled receptor signaling pathway","T cell activation involved in immune response","T cell receptor signaling pathway","adaptive immune response","antigen processing and presentation of endogenous peptide antigen via MHC class I","antigen processing and presentation of exogenous peptide antigen via MHC class I, TAP-dependent","antigen processing and presentation of exogenous peptide antigen via MHC class II","antigen processing and presentation via MHC class Ib","cellular response to interferon-gamma","cellular response to interleukin-1","cellular response to lipopolysaccharide","cellular response to tumor necrosis factor","chemokine-mediated signaling pathway","complement activation, classical pathway","cytokine secretion","defense response to Gram-positive bacterium","defense response to protozoan","defense response to virus","double-strand break repair via break-induced replication","inflammatory response","innate immune response","interferon-gamma-mediated signaling pathway","lymphocyte chemotaxis","macrophage activation","monocyte chemotaxis","negative regulation of interleukin-10 production","negative regulation of viral genome replication","neutrophil chemotaxis","positive regulation of ERK1 and ERK2 cascade","positive regulation of T cell apoptotic process","positive regulation of T cell mediated cytotoxicity","positive regulation of T cell proliferation","positive regulation of defense response to virus by host","positive regulation of inflammatory response","positive regulation of interferon-gamma production","positive regulation of interleukin-1 beta secretion","positive regulation of peptidyl-tyrosine phosphorylation","positive regulation of response to external stimulus","positive regulation of tyrosine phosphorylation of STAT protein","regulation of apoptotic cell clearance","regulation of complement activation","regulation of transcription involved in G1/S transition of mitotic cell cycle","response to interferon-alpha","response to interferon-gamma","type I interferon signaling pathway"],"color.background":["#4169AECC","#313695CC","#E8F6E7CC","#FBFDC4CC","#FDD98ACC","#F46F44CC","#FEFBBACC","#FEE89DCC","#F5FBD0CC","#E2F3F3CC","#F46F44CC","#FDD98ACC","#FEE89DCC","#FEF5B0CC","#E8F6E7CC","#FBFDC4CC","#FEE89DCC","#E8F6E7CC","#E8F6E7CC","#FBA45CCC","#3D5CA7CC","#FDBB6DCC","#A50026CC","#FDB163CC","#D9EFF6CC","#F5FBD0CC","#EFF8DCCC","#B9DFECCC","#FEF5B0CC","#F5FBD0CC","#E8F6E7CC","#CEEAF3CC","#E8F6E7CC","#FEF5B0CC","#F5FBD0CC","#EFF8DCCC","#EFF8DCCC","#E8F6E7CC","#EFF8DCCC","#FEF5B0CC","#F5FBD0CC","#E8F6E7CC","#FEFBBACC","#4169AECC","#F5FBD0CC","#EF633ECC","#FDCF80CC"],"color.highlight":["#4169AE","#313695","#E8F6E7","#FBFDC4","#FDD98A","#F46F44","#FEFBBA","#FEE89D","#F5FBD0","#E2F3F3","#F46F44","#FDD98A","#FEE89D","#FEF5B0","#E8F6E7","#FBFDC4","#FEE89D","#E8F6E7","#E8F6E7","#FBA45C","#3D5CA7","#FDBB6D","#A50026","#FDB163","#D9EFF6","#F5FBD0","#EFF8DC","#B9DFEC","#FEF5B0","#F5FBD0","#E8F6E7","#CEEAF3","#E8F6E7","#FEF5B0","#F5FBD0","#EFF8DC","#EFF8DC","#E8F6E7","#EFF8DC","#FEF5B0","#F5FBD0","#E8F6E7","#FEFBBA","#4169AE","#F5FBD0","#EF633E","#FDCF80"],"color.hover":["#4169AE80","#31369580","#E8F6E780","#FBFDC480","#FDD98A80","#F46F4480","#FEFBBA80","#FEE89D80","#F5FBD080","#E2F3F380","#F46F4480","#FDD98A80","#FEE89D80","#FEF5B080","#E8F6E780","#FBFDC480","#FEE89D80","#E8F6E780","#E8F6E780","#FBA45C80","#3D5CA780","#FDBB6D80","#A5002680","#FDB16380","#D9EFF680","#F5FBD080","#EFF8DC80","#B9DFEC80","#FEF5B080","#F5FBD080","#E8F6E780","#CEEAF380","#E8F6E780","#FEF5B080","#F5FBD080","#EFF8DC80","#EFF8DC80","#E8F6E780","#EFF8DC80","#FEF5B080","#F5FBD080","#E8F6E780","#FEFBBA80","#4169AE80","#F5FBD080","#EF633E80","#FDCF8080"],"color.border":["black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black"],"label":["DNA replication checkpoint","DNA replication initiation","G protein-coupled receptor signaling pathway","T cell activation involved in immune response","T cell receptor signaling pathway","adaptive immune response","antigen processing and presentation of endogenous peptide antigen via MHC class I","antigen processing and presentation of exogenous peptide antigen via MHC class I, TAP-dependent","antigen processing and presentation of exogenous peptide antigen via MHC class II","antigen processing and presentation via MHC class Ib","cellular response to interferon-gamma","cellular response to interleukin-1","cellular response to lipopolysaccharide","cellular response to tumor necrosis factor","chemokine-mediated signaling pathway","complement activation, classical pathway","cytokine secretion","defense response to Gram-positive bacterium","defense response to protozoan","defense response to virus","double-strand break repair via break-induced replication","inflammatory response","innate immune response","interferon-gamma-mediated signaling pathway","lymphocyte chemotaxis","macrophage activation","monocyte chemotaxis","negative regulation of interleukin-10 production","negative regulation of viral genome replication","neutrophil chemotaxis","positive regulation of ERK1 and ERK2 cascade","positive regulation of T cell apoptotic process","positive regulation of T cell mediated cytotoxicity","positive regulation of T cell proliferation","positive regulation of defense response to virus by host","positive regulation of inflammatory response","positive regulation of interferon-gamma production","positive regulation of interleukin-1 beta secretion","positive regulation of peptidyl-tyrosine phosphorylation","positive regulation of response to external stimulus","positive regulation of tyrosine phosphorylation of STAT protein","regulation of apoptotic cell clearance","regulation of complement activation","regulation of transcription involved in G1/S transition of mitotic cell cycle","response to interferon-alpha","response to interferon-gamma","type I interferon signaling pathway"],"x":[0.151980553487874,0.0767475616661304,-0.334764083846429,-0.625383451987635,-0.859461425146719,-0.824839646946179,-0.795251771595235,-1,-0.90729708570826,-0.912349904444768,-0.676678810212797,-0.343512222071411,0.0571453915489684,-0.120074971489706,-0.261471030980644,0.549932787482462,0.785680986289052,0.355650397530235,0.420309946742188,-0.362955572039738,-0.0706345417608725,0.304714270453037,-0.823060426593822,-0.861680697549264,-0.360884795034408,0.693521355036805,-0.07332154729119,0.589756817916478,-0.0970454968425936,-0.154603194128289,-0.225398968013402,0.690879886494096,-0.830350455930248,-0.386862012436543,-0.217445389948658,0.540600734894495,-0.471499370309704,0.948201727826471,1,0.489535408844865,0.921967330483617,0.681585360395911,0.478816139070622,0.259649624520107,-0.446357383381122,-0.92258244278214,-0.710104027559828],"y":[0.733645727754022,0.891642794595309,-0.981631686937787,0.553836927920506,0.761561571170602,0.338019569172141,0.0808782910021966,0.00889785316031033,0.569701985279783,0.113518526716541,-0.346270434154013,-0.595564899189071,-1,-0.462851773609694,-0.813331920141044,-0.166091287649031,0.323815570456768,0.404091389734824,0.254536344466374,0.125842582389407,1,-0.585450096385563,-0.354383739231294,-0.157101804031823,-0.786964042568594,-0.711484557635524,-0.662570274849052,0.756058191476743,0.130858212790201,-0.845085674919037,-0.687532198725645,0.632647352920347,-0.0149408929131903,0.893953313324958,0.330167275938513,-0.613534667534984,0.730838434198475,0.350970693245407,-0.126625871793714,-0.746560897987555,0.0166401986979392,-0.273769877914327,-0.354190603453764,0.865234376877339,-0.114879888798111,-0.303432397258184,-0.0789756698919128]},"edges":{"from":["DNA replication checkpoint","DNA replication checkpoint","DNA replication initiation","DNA replication initiation","G protein-coupled receptor signaling pathway","G protein-coupled receptor signaling pathway","G protein-coupled receptor signaling pathway","T cell activation involved in immune response","T cell activation involved in immune response","T cell receptor signaling pathway","adaptive immune response","adaptive immune response","adaptive immune response","antigen processing and presentation of endogenous peptide antigen via MHC class I","antigen processing and presentation of endogenous peptide antigen via MHC class I","antigen processing and presentation of endogenous peptide antigen via MHC class I","antigen processing and presentation of endogenous peptide antigen via MHC class I","antigen processing and presentation of endogenous peptide antigen via MHC class I","antigen processing and presentation of exogenous peptide antigen via MHC class I, TAP-dependent","antigen processing and presentation of exogenous peptide antigen via MHC class I, TAP-dependent","antigen processing and presentation of exogenous peptide antigen via MHC class I, TAP-dependent","antigen processing and presentation via MHC class Ib","antigen processing and presentation via MHC class Ib","antigen processing and presentation via MHC class Ib","cellular response to interferon-gamma","cellular response to interferon-gamma","cellular response to interferon-gamma","cellular response to interferon-gamma","cellular response to interferon-gamma","cellular response to interleukin-1","cellular response to interleukin-1","cellular response to interleukin-1","cellular response to interleukin-1","cellular response to interleukin-1","cellular response to interleukin-1","cellular response to lipopolysaccharide","cellular response to tumor necrosis factor","chemokine-mediated signaling pathway","chemokine-mediated signaling pathway","chemokine-mediated signaling pathway","chemokine-mediated signaling pathway","complement activation, classical pathway","complement activation, classical pathway","cytokine secretion","defense response to Gram-positive bacterium","defense response to virus","defense response to virus","defense response to virus","inflammatory response","inflammatory response","inflammatory response","inflammatory response","innate immune response","innate immune response","innate immune response","interferon-gamma-mediated signaling pathway","interferon-gamma-mediated signaling pathway","interferon-gamma-mediated signaling pathway","lymphocyte chemotaxis","lymphocyte chemotaxis","lymphocyte chemotaxis","macrophage activation","macrophage activation","monocyte chemotaxis","monocyte chemotaxis","negative regulation of interleukin-10 production","neutrophil chemotaxis","positive regulation of T cell mediated cytotoxicity","positive regulation of T cell proliferation","positive regulation of inflammatory response","positive regulation of peptidyl-tyrosine phosphorylation","regulation of apoptotic cell clearance","response to interferon-alpha","response to interferon-gamma"],"to":["DNA replication initiation","regulation of transcription involved in G1/S transition of mitotic cell cycle","double-strand break repair via break-induced replication","regulation of transcription involved in G1/S transition of mitotic cell cycle","chemokine-mediated signaling pathway","lymphocyte chemotaxis","neutrophil chemotaxis","adaptive immune response","positive regulation of interferon-gamma production","antigen processing and presentation of exogenous peptide antigen via MHC class II","antigen processing and presentation of exogenous peptide antigen via MHC class II","antigen processing and presentation via MHC class Ib","positive regulation of T cell mediated cytotoxicity","antigen processing and presentation of exogenous peptide antigen via MHC class I, TAP-dependent","antigen processing and presentation via MHC class Ib","interferon-gamma-mediated signaling pathway","positive regulation of T cell mediated cytotoxicity","type I interferon signaling pathway","antigen processing and presentation via MHC class Ib","positive regulation of T cell mediated cytotoxicity","type I interferon signaling pathway","interferon-gamma-mediated signaling pathway","positive regulation of T cell mediated cytotoxicity","type I interferon signaling pathway","cellular response to interleukin-1","innate immune response","interferon-gamma-mediated signaling pathway","positive regulation of T cell mediated cytotoxicity","response to interferon-gamma","cellular response to tumor necrosis factor","chemokine-mediated signaling pathway","lymphocyte chemotaxis","monocyte chemotaxis","neutrophil chemotaxis","positive regulation of ERK1 and ERK2 cascade","neutrophil chemotaxis","monocyte chemotaxis","lymphocyte chemotaxis","monocyte chemotaxis","neutrophil chemotaxis","positive regulation of ERK1 and ERK2 cascade","regulation of apoptotic cell clearance","regulation of complement activation","positive regulation of interleukin-1 beta secretion","defense response to protozoan","negative regulation of viral genome replication","positive regulation of defense response to virus by host","type I interferon signaling pathway","monocyte chemotaxis","positive regulation of inflammatory response","positive regulation of response to external stimulus","regulation of complement activation","interferon-gamma-mediated signaling pathway","response to interferon-gamma","type I interferon signaling pathway","positive regulation of T cell mediated cytotoxicity","response to interferon-gamma","type I interferon signaling pathway","monocyte chemotaxis","neutrophil chemotaxis","positive regulation of ERK1 and ERK2 cascade","positive regulation of inflammatory response","positive regulation of response to external stimulus","neutrophil chemotaxis","positive regulation of ERK1 and ERK2 cascade","positive regulation of T cell apoptotic process","positive regulation of ERK1 and ERK2 cascade","type I interferon signaling pathway","positive regulation of interferon-gamma production","positive regulation of response to external stimulus","positive regulation of tyrosine phosphorylation of STAT protein","regulation of complement activation","type I interferon signaling pathway","type I interferon signaling pathway"],"color":["lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey"]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot","physics":false},"manipulation":{"enabled":false},"edges":{"smooth":false},"physics":{"stabilization":false}},"groups":null,"width":null,"height":null,"idselection":{"enabled":false},"byselection":{"enabled":false},"main":null,"submain":null,"footer":null,"background":"rgba(0, 0, 0, 0)","igraphlayout":{"type":"square"}},"evals":[],"jsHooks":[]}</script></code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://federicomarini.github.io" class="external-link">Federico Marini</a>, Annekathrin Ludt.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.0.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

