<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Plot a heatmap of the gene signature on the data — gs_heatmap • GeneTonic</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Plot a heatmap of the gene signature on the data — gs_heatmap"><meta property="og:description" content="Plot a heatmap for the selected gene signature on the provided data, with the possibility to compactly display also DE only genes"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GeneTonic</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.1.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/GeneTonic_manual.html">The `GeneTonic` User's Guide
</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/federicomarini/GeneTonic" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot a heatmap of the gene signature on the data</h1>
    <small class="dont-index">Source: <a href="https://github.com/federicomarini/GeneTonic/blob/HEAD/R/gs_heatmap.R" class="external-link"><code>R/gs_heatmap.R</code></a></small>
    <div class="hidden name"><code>gs_heatmap.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Plot a heatmap for the selected gene signature on the provided data, with the possibility to compactly display also DE only genes</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">gs_heatmap</span><span class="op">(</span></span>
<span>  <span class="va">se</span>,</span>
<span>  <span class="va">res_de</span>,</span>
<span>  <span class="va">res_enrich</span>,</span>
<span>  annotation_obj <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gtl <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  geneset_id <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  genelist <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  FDR <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  de_only <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  cluster_rows <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  cluster_columns <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  center_mean <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  scale_row <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  winsorize_threshold <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  anno_col_info <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  plot_title <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>se</dt>
<dd><p>A <code>SummarizedExperiment</code> object, or an object derived from this class,
such as a <code>DESeqTransform</code> object (variance stabilized transformed data, or
regularized logarithm transformed), in where the transformation has been applied
to make the data more homoscedastic and thus a better fit for visualization.</p></dd>


<dt>res_de</dt>
<dd><p>A <code>DESeqResults</code> object.</p></dd>


<dt>res_enrich</dt>
<dd><p>A <code>data.frame</code> object, storing the result of the functional
enrichment analysis. See more in the main function, <code><a href="GeneTonic.html">GeneTonic()</a></code>, to check the
formatting requirements (a minimal set of columns should be present).</p></dd>


<dt>annotation_obj</dt>
<dd><p>A <code>data.frame</code> object with the feature annotation
information, with at least two columns, <code>gene_id</code> and <code>gene_name</code>.</p></dd>


<dt>gtl</dt>
<dd><p>A <code>GeneTonic</code>-list object, containing in its slots the arguments
specified above: <code>dds</code>, <code>res_de</code>, <code>res_enrich</code>, and <code>annotation_obj</code> - the names
of the list <em>must</em> be specified following the content they are expecting</p></dd>


<dt>geneset_id</dt>
<dd><p>Character specifying the gene set identifier to be plotted</p></dd>


<dt>genelist</dt>
<dd><p>A vector of character strings, specifying the identifiers
contained in the row names of the <code>se</code> input object.</p></dd>


<dt>FDR</dt>
<dd><p>Numeric value, specifying the significance level for thresholding
adjusted p-values. Defaults to 0.05.</p></dd>


<dt>de_only</dt>
<dd><p>Logical, whether to include only differentially expressed genes
in the plot</p></dd>


<dt>cluster_rows</dt>
<dd><p>Logical, determining if rows should be clustered, as
specified by <code><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">ComplexHeatmap::Heatmap()</a></code></p></dd>


<dt>cluster_columns</dt>
<dd><p>Logical, determining if columns should be clustered, as
specified by <code><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">ComplexHeatmap::Heatmap()</a></code></p></dd>


<dt>center_mean</dt>
<dd><p>Logical, whether to perform mean centering on the row-wise</p></dd>


<dt>scale_row</dt>
<dd><p>Logical, whether to standardize by row the expression values</p></dd>


<dt>winsorize_threshold</dt>
<dd><p>Numeric value, to be applied as value to winsorize
the extreme values of the heatmap. Should be a positive number. Defaults to
NULL, which corresponds to not applying any winsorization. Suggested values:
enter 2 or 3 if using row-standardized values (<code>scale_row</code> is TRUE), or visually
inspect the range of the values if using simply mean centered values.</p></dd>


<dt>anno_col_info</dt>
<dd><p>A character vector of names in <code>colData(dds)</code> to use for
decorating the heatmap as annotation.</p></dd>


<dt>plot_title</dt>
<dd><p>Character string, to specify the title of the plot,
displayed over the heatmap. If left to <code>NULL</code> as by default, it tries to use
the information on the geneset identifier provided</p></dd>


<dt>...</dt>
<dd><p>Additional arguments passed to other methods, e.g. in the call to
<code><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">ComplexHeatmap::Heatmap()</a></code></p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A plot returned by the <code><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">ComplexHeatmap::Heatmap()</a></code> function</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"macrophage"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/mikelove/DESeq2" class="external-link">"DESeq2"</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"org.Hs.eg.db"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/AnnotationDbi" class="external-link">"AnnotationDbi"</a></span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># dds object</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"gse"</span>, package <span class="op">=</span> <span class="st">"macrophage"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">dds_macrophage</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSet</a></span><span class="op">(</span><span class="va">gse</span>, design <span class="op">=</span> <span class="op">~</span> <span class="va">line</span> <span class="op">+</span> <span class="va">condition</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> using counts and average transcript lengths from tximeta</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dds_macrophage</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html" class="external-link">substr</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dds_macrophage</span><span class="op">)</span>, <span class="fl">1</span>, <span class="fl">15</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">dds_macrophage</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">estimateSizeFactors</a></span><span class="op">(</span><span class="va">dds_macrophage</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> using 'avgTxLength' from assays(dds), correcting for library size</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">vst_macrophage</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/vst.html" class="external-link">vst</a></span><span class="op">(</span><span class="va">dds_macrophage</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># annotation object</span></span></span>
<span class="r-in"><span><span class="va">anno_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  gene_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dds_macrophage</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  gene_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">mapIds</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>,</span></span>
<span class="r-in"><span>    keys <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dds_macrophage</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    column <span class="op">=</span> <span class="st">"SYMBOL"</span>,</span></span>
<span class="r-in"><span>    keytype <span class="op">=</span> <span class="st">"ENSEMBL"</span></span></span>
<span class="r-in"><span>  <span class="op">)</span>,</span></span>
<span class="r-in"><span>  stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>  row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dds_macrophage</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 'select()' returned 1:many mapping between keys and columns</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># res object</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">res_de_macrophage</span>, package <span class="op">=</span> <span class="st">"GeneTonic"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">res_de</span> <span class="op">&lt;-</span> <span class="va">res_macrophage_IFNg_vs_naive</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># res_enrich object</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">res_enrich_macrophage</span>, package <span class="op">=</span> <span class="st">"GeneTonic"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">res_enrich</span> <span class="op">&lt;-</span> <span class="fu"><a href="shake_topGOtableResult.html">shake_topGOtableResult</a></span><span class="op">(</span><span class="va">topgoDE_macrophage_IFNg_vs_naive</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 500 gene sets in `topGOtableResult` object.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting for usage in GeneTonic...</span>
<span class="r-in"><span><span class="va">res_enrich</span> <span class="op">&lt;-</span> <span class="fu"><a href="get_aggrscores.html">get_aggrscores</a></span><span class="op">(</span><span class="va">res_enrich</span>, <span class="va">res_de</span>, <span class="va">anno_df</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">gs_heatmap</span><span class="op">(</span><span class="va">vst_macrophage</span>,</span></span>
<span class="r-in"><span>  <span class="va">res_de</span>,</span></span>
<span class="r-in"><span>  <span class="va">res_enrich</span>,</span></span>
<span class="r-in"><span>  <span class="va">anno_df</span>,</span></span>
<span class="r-in"><span>  geneset_id <span class="op">=</span> <span class="va">res_enrich</span><span class="op">$</span><span class="va">gs_id</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span></span>
<span class="r-in"><span>  cluster_columns <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>  anno_col_info <span class="op">=</span> <span class="st">"condition"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="gs_heatmap-1.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://federicomarini.github.io" class="external-link">Federico Marini</a>, Annekathrin Ludt.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

